pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -936,9 +936,31 @@ func findRule(rules []network.LoadBalancingRule, rule network.LoadBalancingRule)
 
 func findSecurityRule(rules []network.SecurityRule, rule network.SecurityRule) bool {
 	for _, existingRule := range rules {
-		if strings.EqualFold(*existingRule.Name, *rule.Name) {
-			return true
+		if !strings.EqualFold(*existingRule.Name, *rule.Name) {
+			continue
+		}
+		if existingRule.Protocol != rule.Protocol {
+			continue
+		}
+		if !strings.EqualFold(*existingRule.SourcePortRange, *rule.SourcePortRange) {
+			continue
+		}
+		if !strings.EqualFold(*existingRule.DestinationPortRange, *rule.DestinationPortRange) {
+			continue
+		}
+		if !strings.EqualFold(*existingRule.SourceAddressPrefix, *rule.SourceAddressPrefix) {
+			continue
+		}
+		if !strings.EqualFold(*existingRule.DestinationAddressPrefix, *rule.DestinationAddressPrefix) {
+			continue
+		}
+		if existingRule.Access != rule.Access {
+			continue
+		}
+		if existingRule.Direction != rule.Direction {
+			continue
```

Comment:
Some Azure regions support "augmented security rules" which permit more than one destination address in a rule (https://azure.microsoft.com/en-au/updates/public-preview-features-for-nsgs/).  This is part of the 11.x Azure SDK, which is in master as of https://github.com/kubernetes/kubernetes/commit/14e134cec33960abc87dfe452ac0836804f91ec0.  It might be a good idea to check `rule.DestinationAddressPrefixes` as well as `DestinationAddressPrefix` - we won't hit this with unmodified rules in current master but it could happen if someone modifies a rule through the portal, and it will be used within Kubernetes if #55658 is accepted.

Reply:
hmm I think that additional check can be done in follow-up PR? As for this PR, I think if makes sense to do the check on only controlled properties that are explicitly set while generating security rules. (at least for this branch).